enhancements/dns/plugins.md

```diff
@@ -97,24 +98,28 @@ resolution, it restricts how cluster operators manage DNS name queries.
 
 ## Proposal
 
-`PluginConfig` defines one or more CoreDNS plugins that can be configured. If a plugin is not defined, it will use a
-default configuration provided by the cluster-dns-operator. A plugin is represented as a slice when multiple instances
-of the plugin can be expressed within the DNS configuration.
+`Server` defines the schema for a server that runs per instance of CoreDNS. A `Server` must have a `Name` that is unique
+among `Servers`. `Zones` defines one or more subdomains to match for name queries to be forwarded to `Upstreams`.
+`ForwardPlugin` provides options for configuring the forward plugin configuration. A `Server` listens on port `5353`,
+which can not be configured at this time. If a `Server` is not specified, the cluster-dns-operator will generate the
+previously depicted ConfigMap.
 
 ```go
-type PluginConfig struct {
-    ForwardPlugins []ForwardPlugin `json:"forwardPlugin"`
+type Server struct {
+    Name string `json:"name"`
+    Zones []string `json:"zones"`
+    ForwardPlugin ForwardPlugin `json:"forwardPlugin"`
     // additional future plugins
 }
 ```
 
-The proposed `ForwardPlugin` type defines a schema for associating one or more `Nameservers` to a DNS subdomain
-identified by `Domain`. `Nameservers` are responsible for resolving name queries for `Domain`. `Policy` provides a
-mechanism for distinguishing where to forward DNS messages when `Nameservers` consists of more than one nameserver:
+`ForwardPlugin` defines a schema for configuring the CoreDNS forward plugin. `Upstreams` is a list of resolvers to
+forward name queries for subdomains specified by `Zones`. `Policy` provides a mechanism for selecting an upstream
+resolver when `Upstreams` consists of more than one resolver. If a policy is not specified, the "Random" policy will be
+used which implements random upstream selection.
```

Comment:
The story below does not use `Policy`; can we omit it for now?

Reply:
It has been removed.